src/Components/ServiceScene/LineFilterScene.tsx

```diff
@@ -66,11 +67,17 @@ export class LineFilterScene extends SceneObjectBase<LineFilterState> {
   };
 
   onCaseSensitiveToggle = (newState: 'sensitive' | 'insensitive') => {
+    const caseSensitive = newState === 'sensitive';
+
+    // Set value to scene state
     this.setState({
-      caseSensitive: newState === 'sensitive',
+      caseSensitive,
     });
 
-    this.updateFilter(this.state.lineFilter);
+    // Set value in local storage
+    setLineFilterCase(caseSensitive);
+
+    this.updateFilter(this.state.lineFilter, false);
```

Comment:
Just FYI this call to updateFilter() used to be debounced.

Reply:
Yeah I don't think there's a reason to debounce a button press toggle, plus it meant if you refreshed before the debounce the UI state wouldn't get added to local storage.